packages/frontend/src/utils.ts

```diff
@@ -17,3 +17,17 @@ export function useTimeout(timeout: number) {
 
   return value;
 }
+
+export function useNamedInputState<T>(initialState: T) {
+  const [namedInputState, setState] = useState<T>(initialState);
+
+  const setNamedInputState = (event: any) => {
+    const { target } = event;
+    setState({
+      ...namedInputState,
+      [target.name]: target.value,
+    });
+  };
+
+  return { namedInputState, setNamedInputState };
```

Comment:
se gli fai ritornare un array come setState è un po' meglio visto che puoi permettere la rinonominazione ma vbb

Reply:
In effetti non avevo mai capito perché tornavano Array, good to know🤔